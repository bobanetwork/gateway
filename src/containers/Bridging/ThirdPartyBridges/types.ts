export interface IThirdPartyBridge {
  name: string
  icon?: any
  iconLight?: any
  iconDark?: any
  link?: string
}
