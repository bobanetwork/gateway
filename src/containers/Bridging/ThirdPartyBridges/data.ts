import BanxaLogo from 'assets/images/bridges/banxa.svg'
import BeamerLogo from 'assets/images/bridges/beamer.png'
import BoringDaoLogo from 'assets/images/bridges/Boringdao.png'
import ChainswapLogo from 'assets/images/bridges/chainswap.png'
import ConnextLogo from 'assets/images/bridges/connext.png'
import LayerswapLogo from 'assets/images/bridges/layerswap.png'
import MultichainLogo from 'assets/images/bridges/multichain.png'
import RangoExchangeLogo from 'assets/images/bridges/rango_exchange.png'
import SynapseLogo from 'assets/images/bridges/synapse.png'

import CelerLogoDark from 'assets/images/bridges/dark/dark_celer.png'
import RubicExchangeLogoDark from 'assets/images/bridges/dark/dark_rubic_exchange.png'
import ViaProtocolLogoDark from 'assets/images/bridges/dark/dark_via_protocol.png'
import CelerLogoLight from 'assets/images/bridges/light/light_celer.png'
import RubicExchangeLogoLight from 'assets/images/bridges/light/light_rubic_exchange.png'
import ViaProtocolLogoLight from 'assets/images/bridges/light/light_via_protocol.png'

export interface IBridges {
  name: string
  icon?: any
  iconLight?: any
  iconDark?: any
  link?: string
}

export const bobaBridges: IBridges[] = [
  {
    name: 'Banxa',
    icon: BanxaLogo,
    link: 'https://boba.banxa.com/',
  },
  {
    name: 'Beamer Bridge',
    icon: BeamerLogo,
    link: 'https://boba.network/project/beamer-bridge/',
  },
  {
    name: 'BoringDAO',
    icon: BoringDaoLogo,
    link: 'https://boba.network/project/boringdao/',
  },
  {
    name: 'Celer',
    icon: null,
    iconLight: CelerLogoLight,
    iconDark: CelerLogoDark,
    link: 'https://boba.network/project/celer/',
  },
  {
    name: 'Chainswap',
    icon: ChainswapLogo,
    link: 'https://boba.network/project/chainswap/',
  },
  {
    name: 'Connext',
    icon: ConnextLogo,
    link: 'https://boba.network/project/connext/',
  },
  {
    name: 'Layerswap',
    icon: LayerswapLogo,
    link: 'https://boba.network/project/layerswap-io/',
  },
  {
    name: 'Multichain',
    icon: MultichainLogo,
    link: 'https://boba.network/project/multichain/',
  },
  {
    name: 'Rango Exchange',
    icon: RangoExchangeLogo,
    link: 'https://boba.network/project/rango-exchange/',
  },
  {
    name: 'Rubic Exchange',
    icon: null,
    iconLight: RubicExchangeLogoLight,
    iconDark: RubicExchangeLogoDark,
    link: 'https://boba.network/project/rubic-exchange/',
  },
  {
    name: 'Synapse',
    icon: SynapseLogo,
    link: 'https://boba.network/project/synapse/',
  },
  {
    name: 'Via Protocol',
    icon: null,
    iconLight: ViaProtocolLogoLight,
    iconDark: ViaProtocolLogoDark,
    link: 'https://boba.network/project/via-protocol/',
  },
]
