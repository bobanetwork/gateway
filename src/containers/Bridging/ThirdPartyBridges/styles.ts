import { Typography } from 'components/global'
import styled, { css } from 'styled-components'

export const BridgeItem = styled.a`
  cursor: pointer;
  padding: 16px;
  display: flex;
  align-items: center;
  justify-content: flex-start;
  gap: 8px;
  min-height: 64px;
  border-radius: 12px;
  border: 1px solid
    ${({ theme: { colors, name } }) =>
      name === 'light' ? colors.gray[500] : colors.gray[300]};
  color: ${(props) => props.theme.color};
  text-decoration: none;
  &:hover {
    background: ${({ theme: { colors, name } }) =>
      name === 'light' ? colors.gray[500] : colors.gray[300]};
  }
`
export const BridgeIcon = styled.div`
  height: 32px;
  width: 32px;
  border-radius: 50%;
`
export const BridgeLabel = styled(Typography).attrs({
  variant: 'title',
})`
  flex: 1;
  line-height: normal;
  text-transform: capitalize;
`

export const ThirdPartyBridgesContainer = styled.div`
  display: flex;
  flex-direction: column;
  gap: 14px;
  width: 100%;
`
export const BridgeList = styled.div.withConfig({
  shouldForwardProp: (prop) => !['empty'].includes(prop),
})<{ empty?: boolean }>`
  display: flex;
  flex-direction: column;
  gap: 8px;
  width: 100%;
  height: 320px;
  overflow-y: scroll;

  ${({ empty, theme: { name, colors } }) =>
    empty &&
    css`
      display: grid;
      justify-content: center;
      align-content: center;
      color: ${name === 'light' ? colors.gray[700] : colors.gray[100]};
    `}
`
