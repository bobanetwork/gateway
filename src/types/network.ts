export type NetworkSwitchState =
  | 'idle'
  | 'confirming'
  | 'switching'
  | 'success'
  | 'failed'
  | 'chain-not-found'

export interface NetworkSwitchError {
  code?: number;
  message: string;
}
