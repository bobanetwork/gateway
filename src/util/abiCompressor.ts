/**
 * Script to prepare the human readable abi from the json.
 * based on the functions used out of the abi in network service.
 *
 * to prepare abi execute like
 * ```sh
 *  node ./util/abiCompressor.ts
 * ```
 *
 *
 */

import { ethers } from 'ethers'
import * as fs from 'fs'

const smart_contract_abi_paths = [
  './node_modules/@bobanetwork/core_contracts/artifacts/contracts/L2/messaging/L2StandardBridge.sol/L2StandardBridge.json',
  './node_modules/@bobanetwork/core_contracts/artifacts/contracts/standards/L2StandardERC20.sol/L2StandardERC20.json',
  './node_modules/@bobanetwork/core_contracts/artifacts/contracts/L2/predeploys/OVM_GasPriceOracle.sol/OVM_GasPriceOracle.json',
  './node_modules/@bobanetwork/contracts/artifacts/contracts/DiscretionaryExitFee.sol/DiscretionaryExitFee.json',
  './node_modules/@bobanetwork/contracts/artifacts/contracts/LP/L1LiquidityPool.sol/L1LiquidityPool.json',
  './node_modules/@bobanetwork/contracts/artifacts/contracts/Teleportation.sol/Teleportation.json',
  './node_modules/@bobanetwork/contracts/artifacts/contracts/LP/L2LiquidityPool.sol/L2LiquidityPool.json',
  './node_modules/@bobanetwork/contracts/artifacts/contracts/BobaFixedSavings.sol/BobaFixedSavings.json',
  './node_modules/@bobanetwork/contracts/artifacts/contracts/DAO/governance-token/BOBA.sol/BOBA.json',
  './node_modules/@bobanetwork/contracts/artifacts/contracts/DAO/governance/GovernorBravoDelegate.sol/GovernorBravoDelegate.json',
  './node_modules/@bobanetwork/contracts/artifacts/contracts/DAO/governance/GovernorBravoDelegator.sol/GovernorBravoDelegator.json',
  './node_modules/@bobanetwork/contracts/artifacts/contracts/L2BillingContract.sol/L2BillingContract.json',
  './src/deployment/contracts/L1ERC20.json',
  './src/deployment/contracts/NFTMonsterV2.json',
  './src/deployment/contracts/AuthenticatedFaucet.json',
  './src/deployment/contracts/WAGMIv0.json',
  './src/deployment/contracts/WAGMIv1.json',
  './src/deployment/contracts/crosschain/EthBridge.json',
  './src/deployment/contracts/crosschain/L2StandardERC20.json',
  './src/deployment/contracts/crosschain/LZEndpointMock.json',
]

const networkServicesStr = fs.readFileSync(
  './src/services/networkService.ts',
  'utf8'
)

const compress = () => {
  for (const path of smart_contract_abi_paths) {
    const rawData = fs.readFileSync(path, 'utf8')
    const abiObject = JSON.parse(rawData)
    const name = abiObject.contractName
    let abi = abiObject.abi
    abi = abi.filter((component) => {
      if (component['type'] === 'function' && component['name']) {
        if (networkServicesStr.includes(component['name'])) {
          return true
        }
        return false
      }
    })
    const abiInterface = new ethers.utils.Interface(abi)
    const humanReadableABI = abiInterface.format(ethers.utils.FormatTypes.full)
    const stringifiedHumanReadableABI = JSON.stringify(humanReadableABI)
    const result = `const ${name}ABI = ${stringifiedHumanReadableABI}\n export default ${name}ABI`
    fs.writeFile(`./src/services/abi/${name}.abi.ts`, result, () => {
      console.log(`wrote ${name}`)
    })
  }
}

;(() => {
  compress()
})()
