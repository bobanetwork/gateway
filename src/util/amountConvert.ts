/*
Copyright 2021-present Boba Network.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

     http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License. */

// we use BigNumber here for decimal support
import BigNumber from 'bignumber.js'
import { BigNumberish } from 'ethers'

export const logAmount = (
  amount: BigNumberish,
  power: number,
  truncate = 0
) => {
  // toString as some methods seem to supply BigNumber
  const x = new BigNumber(amount?.toString())
  const exp = new BigNumber(10).pow(power)

  const calculated = x.div(exp)

  if (truncate > 0) {
    return calculated.toFixed(truncate)
  } else {
    return calculated.toFixed()
  }
}

/*Takes a value such as 3.92 and converts it into
a BigNumber in wei

Duplicates

ethers.utils.parseUnits( valueString , decimalsOrUnitName )   =>   BigNumber
*/

export const powAmount = (amount: number, decimals: number) => {
  const x = new BigNumber(amount)
  const exp = new BigNumber(10).pow(decimals)

  const calculated = x.multipliedBy(exp)
  return calculated.toFixed(0)
}

/* more clearly named version of this */
export const toWei_String = (amount: number, decimals: number) => {
  const x = new BigNumber(amount)
  const exp = new BigNumber(10).pow(decimals)

  const calculated = x.multipliedBy(exp)
  return calculated.toFixed(0)
}

const lookupMap = {
  ETH: { provider: 'ethereum' },
  BOBA: { provider: 'boba-network' },
  OLO: { provider: 'oolongswap' },
  OMG: { provider: 'omisego' },
  USDC: { provider: 'usd-coin' },
  BNB: { provider: 'binancecoin' },
  tBNB: { provider: 'binancecoin' },
}

export const amountToUsd = (amount: number, lookupPrice, token) => {
  const { symbol } = token
  if (!symbol) {
    return 0
  }
  const currencyInfo = lookupMap[symbol]
  if (!!currencyInfo && !!lookupPrice[currencyInfo.provider]) {
    return amount * lookupPrice[currencyInfo.provider].usd
  } else if (!!lookupPrice[symbol.toLowerCase()]) {
    return amount * lookupPrice[symbol.toLowerCase()].usd
  } else {
    return 0
  }
}

export const formatLargeNumber = (num: number) => {
  const exp = Math.floor(Math.log10(Math.abs(num)) / 3)
  const scaledNum = Math.abs(num) / Math.pow(10, exp * 3)
  const formattedNumber = scaledNum.toFixed(2)
  const exponent = ['', 'k', 'M', 'B', 'T'][exp] || ''
  return isNaN(Number(formattedNumber)) ? '0' : formattedNumber + exponent
}
