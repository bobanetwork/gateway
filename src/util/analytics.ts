import ReactGA from 'react-ga4'
import { GA4_MEASUREMENT_ID } from './constant'

export const init = () => {
  if (window.location.hostname === 'localhost') {
    return null
  }
  if (GA4_MEASUREMENT_ID) {
    // init if the GA4 Measurement Id is available.
    ReactGA.initialize(GA4_MEASUREMENT_ID?.toString())
  }
}

export const sendPageView = (path: string) => {
  ReactGA.send({
    hitType: 'pageview',
    page: path,
    title: path,
  })
}

export const trackClick = (category: string, action: string, label: string) => {
  ReactGA.event({
    category,
    action,
    label,
  })
}

export default {
  init,
  sendPageView,
  trackClick,
}
