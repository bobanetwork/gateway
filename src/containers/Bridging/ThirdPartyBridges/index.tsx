import React, { FC } from 'react'
import { useSelector } from 'react-redux'
import { selectActiveNetwork, selectActiveNetworkType } from 'selectors'

import { Typography } from 'components/global'
import useThirdPartyBridges from 'hooks/useThirdPartyBridges'
import { useTheme } from 'styled-components'
import { Network, NetworkType } from 'util/network/network.util'
import { SectionLabel } from '../chain/styles'
import {
  BridgeIcon,
  BridgeItem,
  BridgeLabel,
  BridgeList,
  ThirdPartyBridgesContainer,
} from './styles'
import { IThirdPartyBridge } from './types'

const ThirdPartyBridges: FC = () => {
  const networkType = useSelector(selectActiveNetworkType())
  const network = useSelector(selectActiveNetwork())
  const { loading, bridges } = useThirdPartyBridges<IThirdPartyBridge[]>()

  const theme: any = useTheme()

  if (networkType !== NetworkType.MAINNET) {
    return <></>
  }

  return (
    <ThirdPartyBridgesContainer>
      <SectionLabel variant="body2">Third party bridges</SectionLabel>
      {network !== Network.ETHEREUM ? (
        <BridgeList empty>
          <Typography variant="body1">No bridges available</Typography>
        </BridgeList>
      ) : (
        <BridgeList>
          {!loading ? (
            Array.isArray(bridges) &&
            bridges.map((bridge: IThirdPartyBridge) => (
              <BridgeItem
                href={bridge.link}
                target="_blank"
                rel="noopener noreferrer"
                key={bridge.name}
                data-testid="bridge-item"
              >
                <BridgeIcon>
                  <img
                    src={
                      bridge.icon
                        ? bridge.icon
                        : theme.name === 'light'
                          ? bridge.iconLight
                          : bridge.iconDark
                    }
                    alt={`${bridge.name} ${
                      bridge.icon ? '' : theme.name + '-'
                    }logo`}
                    width="32px"
                    height="32px"
                  />
                </BridgeIcon>
                <BridgeLabel>{bridge.name}</BridgeLabel>
              </BridgeItem>
            ))
          ) : (
            <Typography variant="body1">Loading ...</Typography>
          )}
        </BridgeList>
      )}
    </ThirdPartyBridgesContainer>
  )
}

export default ThirdPartyBridges
