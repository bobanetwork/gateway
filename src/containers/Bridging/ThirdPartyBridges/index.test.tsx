import React from 'react'
import { render, screen } from '@testing-library/react'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import { Network, NetworkType } from 'util/network/network.util'
import CustomThemeProvider from 'themes'
import ThirdPartyBridges from '.'
import { mockedInitialState } from 'util/tests'

const mockStore = configureStore()

const renderThirdPartyBridges = ({ options = {} }: any) => {
  return render(
    <Provider
      store={mockStore({
        ...mockedInitialState,
        ...options,
      })}
    >
      <CustomThemeProvider>
        <ThirdPartyBridges />
      </CustomThemeProvider>
    </Provider>
  )
}

describe('3rd Party Bridges', () => {
  test('should match snapshot when network is ETH Mainnet', () => {
    const { asFragment } = renderThirdPartyBridges({})
    expect(asFragment()).toMatchSnapshot()
  })

  test('should match snapshot when network is BNB Mainnet', () => {
    const { asFragment } = renderThirdPartyBridges({
      options: {
        network: {
          activeNetwork: Network.BNB,
          activeNetworkType: NetworkType.MAINNET,
        },
      },
    })
    expect(asFragment()).toMatchSnapshot()
  })

  test('should match snapshot when network is TESTNET', () => {
    const { asFragment } = renderThirdPartyBridges({
      options: {
        network: {
          activeNetwork: Network.ETHEREUM,
          activeNetworkType: NetworkType.TESTNET,
        },
      },
    })
    expect(asFragment()).toMatchSnapshot()
  })

  test('should render icons correctly for light mode', () => {
    const { asFragment } = renderThirdPartyBridges({
      options: {
        ui: {
          theme: 'light',
        },
      },
    })
    expect(asFragment()).toMatchSnapshot()

    expect(screen.getAllByAltText('light-logo', { exact: false }).length).toBe(
      3
    )
  })

  test('should render bridge list correctly', () => {
    renderThirdPartyBridges({})
    expect(screen.getAllByTestId('bridge-item').length).toBe(8)
    expect(screen.getByText('Banxa')).toBeInTheDocument()
    expect(screen.getByText('BoringDAO')).toBeInTheDocument()
    expect(screen.getByText('Celer')).toBeInTheDocument()
    expect(screen.getByText('Chainswap')).toBeInTheDocument()
    expect(screen.getByText('Rango Exchange')).toBeInTheDocument()
    expect(screen.getByText('Rubic Exchange')).toBeInTheDocument()
    expect(screen.getByText('Synapse')).toBeInTheDocument()
    expect(screen.getByText('Via Protocol')).toBeInTheDocument()
  })
})
