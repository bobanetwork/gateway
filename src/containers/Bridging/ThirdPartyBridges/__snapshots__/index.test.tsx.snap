// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`3rd Party Bridges should match snapshot when network is BNB Mainnet 1`] = `
<DocumentFragment>
  <div
    class="sc-iHGNWf jDvRrm"
  >
    <p
      class="sc-jXbUNg jCIOIT sc-jsJBEP hgyjkx"
    >
      Third party bridges
    </p>
    <div
      class="sc-dtBdUo lasFkC"
    >
      <p
        class="sc-jXbUNg gzUJEq"
      >
        No bridges available
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`3rd Party Bridges should match snapshot when network is ETH Mainnet 1`] = `
<DocumentFragment>
  <div
    class="sc-iHGNWf jDvRrm"
  >
    <p
      class="sc-jXbUNg jCIOIT sc-jsJBEP hgyjkx"
    >
      Third party bridges
    </p>
    <div
      class="sc-dtBdUo csjtne"
    >
      <a
        class="sc-eeDRCY dndUuq"
        data-testid="bridge-item"
        href="https://boba.banxa.com/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          class="sc-koXPp fZGqGt"
        >
          <img
            alt="Banxa logo"
            height="32px"
            src="banxa.svg"
            width="32px"
          />
        </div>
        <p
          class="sc-jXbUNg hMVWVX sc-bmzYkS jfckxU"
        >
          Banxa
        </p>
      </a>
      <a
        class="sc-eeDRCY dndUuq"
        data-testid="bridge-item"
        href="https://boba.network/project/beamer-bridge/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          class="sc-koXPp fZGqGt"
        >
          <img
            alt="Beamer Bridge logo"
            height="32px"
            src="beamer.png"
            width="32px"
          />
        </div>
        <p
          class="sc-jXbUNg hMVWVX sc-bmzYkS jfckxU"
        >
          Beamer Bridge
        </p>
      </a>
      <a
        class="sc-eeDRCY dndUuq"
        data-testid="bridge-item"
        href="https://boba.network/project/boringdao/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          class="sc-koXPp fZGqGt"
        >
          <img
            alt="BoringDAO logo"
            height="32px"
            src="Boringdao.png"
            width="32px"
          />
        </div>
        <p
          class="sc-jXbUNg hMVWVX sc-bmzYkS jfckxU"
        >
          BoringDAO
        </p>
      </a>
      <a
        class="sc-eeDRCY dndUuq"
        data-testid="bridge-item"
        href="https://boba.network/project/celer/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          class="sc-koXPp fZGqGt"
        >
          <img
            alt="Celer dark-logo"
            height="32px"
            src="dark_celer.png"
            width="32px"
          />
        </div>
        <p
          class="sc-jXbUNg hMVWVX sc-bmzYkS jfckxU"
        >
          Celer
        </p>
      </a>
      <a
        class="sc-eeDRCY dndUuq"
        data-testid="bridge-item"
        href="https://boba.network/project/chainswap/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          class="sc-koXPp fZGqGt"
        >
          <img
            alt="Chainswap logo"
            height="32px"
            src="chainswap.png"
            width="32px"
          />
        </div>
        <p
          class="sc-jXbUNg hMVWVX sc-bmzYkS jfckxU"
        >
          Chainswap
        </p>
      </a>
      <a
        class="sc-eeDRCY dndUuq"
        data-testid="bridge-item"
        href="https://boba.network/project/connext/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          class="sc-koXPp fZGqGt"
        >
          <img
            alt="Connext logo"
            height="32px"
            src="connext.png"
            width="32px"
          />
        </div>
        <p
          class="sc-jXbUNg hMVWVX sc-bmzYkS jfckxU"
        >
          Connext
        </p>
      </a>
      <a
        class="sc-eeDRCY dndUuq"
        data-testid="bridge-item"
        href="https://boba.network/project/layerswap-io/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          class="sc-koXPp fZGqGt"
        >
          <img
            alt="Layerswap logo"
            height="32px"
            src="layerswap.png"
            width="32px"
          />
        </div>
        <p
          class="sc-jXbUNg hMVWVX sc-bmzYkS jfckxU"
        >
          Layerswap
        </p>
      </a>
      <a
        class="sc-eeDRCY dndUuq"
        data-testid="bridge-item"
        href="https://boba.network/project/multichain/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          class="sc-koXPp fZGqGt"
        >
          <img
            alt="Multichain logo"
            height="32px"
            src="multichain.png"
            width="32px"
          />
        </div>
        <p
          class="sc-jXbUNg hMVWVX sc-bmzYkS jfckxU"
        >
          Multichain
        </p>
      </a>
      <a
        class="sc-eeDRCY dndUuq"
        data-testid="bridge-item"
        href="https://boba.network/project/rango-exchange/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          class="sc-koXPp fZGqGt"
        >
          <img
            alt="Rango Exchange logo"
            height="32px"
            src="rango_exchange.png"
            width="32px"
          />
        </div>
        <p
          class="sc-jXbUNg hMVWVX sc-bmzYkS jfckxU"
        >
          Rango Exchange
        </p>
      </a>
      <a
        class="sc-eeDRCY dndUuq"
        data-testid="bridge-item"
        href="https://boba.network/project/rubic-exchange/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          class="sc-koXPp fZGqGt"
        >
          <img
            alt="Rubic Exchange dark-logo"
            height="32px"
            src="dark_rubic_exchange.png"
            width="32px"
          />
        </div>
        <p
          class="sc-jXbUNg hMVWVX sc-bmzYkS jfckxU"
        >
          Rubic Exchange
        </p>
      </a>
      <a
        class="sc-eeDRCY dndUuq"
        data-testid="bridge-item"
        href="https://boba.network/project/synapse/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          class="sc-koXPp fZGqGt"
        >
          <img
            alt="Synapse logo"
            height="32px"
            src="synapse.png"
            width="32px"
          />
        </div>
        <p
          class="sc-jXbUNg hMVWVX sc-bmzYkS jfckxU"
        >
          Synapse
        </p>
      </a>
      <a
        class="sc-eeDRCY dndUuq"
        data-testid="bridge-item"
        href="https://boba.network/project/via-protocol/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          class="sc-koXPp fZGqGt"
        >
          <img
            alt="Via Protocol dark-logo"
            height="32px"
            src="dark_via_protocol.png"
            width="32px"
          />
        </div>
        <p
          class="sc-jXbUNg hMVWVX sc-bmzYkS jfckxU"
        >
          Via Protocol
        </p>
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`3rd Party Bridges should match snapshot when network is TESTNET 1`] = `<DocumentFragment />`;

exports[`3rd Party Bridges should render icons correctly for light mode 1`] = `
<DocumentFragment>
  <div
    class="sc-iHGNWf jDvRrm"
  >
    <p
      class="sc-jXbUNg jCIOIT sc-jsJBEP jnmkdr"
    >
      Third party bridges
    </p>
    <div
      class="sc-dtBdUo csjtne"
    >
      <a
        class="sc-eeDRCY ezvrQC"
        data-testid="bridge-item"
        href="https://boba.banxa.com/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          class="sc-koXPp fZGqGt"
        >
          <img
            alt="Banxa logo"
            height="32px"
            src="banxa.svg"
            width="32px"
          />
        </div>
        <p
          class="sc-jXbUNg hMVWVX sc-bmzYkS jfckxU"
        >
          Banxa
        </p>
      </a>
      <a
        class="sc-eeDRCY ezvrQC"
        data-testid="bridge-item"
        href="https://boba.network/project/beamer-bridge/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          class="sc-koXPp fZGqGt"
        >
          <img
            alt="Beamer Bridge logo"
            height="32px"
            src="beamer.png"
            width="32px"
          />
        </div>
        <p
          class="sc-jXbUNg hMVWVX sc-bmzYkS jfckxU"
        >
          Beamer Bridge
        </p>
      </a>
      <a
        class="sc-eeDRCY ezvrQC"
        data-testid="bridge-item"
        href="https://boba.network/project/boringdao/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          class="sc-koXPp fZGqGt"
        >
          <img
            alt="BoringDAO logo"
            height="32px"
            src="Boringdao.png"
            width="32px"
          />
        </div>
        <p
          class="sc-jXbUNg hMVWVX sc-bmzYkS jfckxU"
        >
          BoringDAO
        </p>
      </a>
      <a
        class="sc-eeDRCY ezvrQC"
        data-testid="bridge-item"
        href="https://boba.network/project/celer/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          class="sc-koXPp fZGqGt"
        >
          <img
            alt="Celer light-logo"
            height="32px"
            src="light_celer.png"
            width="32px"
          />
        </div>
        <p
          class="sc-jXbUNg hMVWVX sc-bmzYkS jfckxU"
        >
          Celer
        </p>
      </a>
      <a
        class="sc-eeDRCY ezvrQC"
        data-testid="bridge-item"
        href="https://boba.network/project/chainswap/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          class="sc-koXPp fZGqGt"
        >
          <img
            alt="Chainswap logo"
            height="32px"
            src="chainswap.png"
            width="32px"
          />
        </div>
        <p
          class="sc-jXbUNg hMVWVX sc-bmzYkS jfckxU"
        >
          Chainswap
        </p>
      </a>
      <a
        class="sc-eeDRCY ezvrQC"
        data-testid="bridge-item"
        href="https://boba.network/project/connext/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          class="sc-koXPp fZGqGt"
        >
          <img
            alt="Connext logo"
            height="32px"
            src="connext.png"
            width="32px"
          />
        </div>
        <p
          class="sc-jXbUNg hMVWVX sc-bmzYkS jfckxU"
        >
          Connext
        </p>
      </a>
      <a
        class="sc-eeDRCY ezvrQC"
        data-testid="bridge-item"
        href="https://boba.network/project/layerswap-io/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          class="sc-koXPp fZGqGt"
        >
          <img
            alt="Layerswap logo"
            height="32px"
            src="layerswap.png"
            width="32px"
          />
        </div>
        <p
          class="sc-jXbUNg hMVWVX sc-bmzYkS jfckxU"
        >
          Layerswap
        </p>
      </a>
      <a
        class="sc-eeDRCY ezvrQC"
        data-testid="bridge-item"
        href="https://boba.network/project/multichain/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          class="sc-koXPp fZGqGt"
        >
          <img
            alt="Multichain logo"
            height="32px"
            src="multichain.png"
            width="32px"
          />
        </div>
        <p
          class="sc-jXbUNg hMVWVX sc-bmzYkS jfckxU"
        >
          Multichain
        </p>
      </a>
      <a
        class="sc-eeDRCY ezvrQC"
        data-testid="bridge-item"
        href="https://boba.network/project/rango-exchange/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          class="sc-koXPp fZGqGt"
        >
          <img
            alt="Rango Exchange logo"
            height="32px"
            src="rango_exchange.png"
            width="32px"
          />
        </div>
        <p
          class="sc-jXbUNg hMVWVX sc-bmzYkS jfckxU"
        >
          Rango Exchange
        </p>
      </a>
      <a
        class="sc-eeDRCY ezvrQC"
        data-testid="bridge-item"
        href="https://boba.network/project/rubic-exchange/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          class="sc-koXPp fZGqGt"
        >
          <img
            alt="Rubic Exchange light-logo"
            height="32px"
            src="light_rubic_exchange.png"
            width="32px"
          />
        </div>
        <p
          class="sc-jXbUNg hMVWVX sc-bmzYkS jfckxU"
        >
          Rubic Exchange
        </p>
      </a>
      <a
        class="sc-eeDRCY ezvrQC"
        data-testid="bridge-item"
        href="https://boba.network/project/synapse/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          class="sc-koXPp fZGqGt"
        >
          <img
            alt="Synapse logo"
            height="32px"
            src="synapse.png"
            width="32px"
          />
        </div>
        <p
          class="sc-jXbUNg hMVWVX sc-bmzYkS jfckxU"
        >
          Synapse
        </p>
      </a>
      <a
        class="sc-eeDRCY ezvrQC"
        data-testid="bridge-item"
        href="https://boba.network/project/via-protocol/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          class="sc-koXPp fZGqGt"
        >
          <img
            alt="Via Protocol light-logo"
            height="32px"
            src="light_via_protocol.png"
            width="32px"
          />
        </div>
        <p
          class="sc-jXbUNg hMVWVX sc-bmzYkS jfckxU"
        >
          Via Protocol
        </p>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
